src/javascript/app/base/logged_in.js

```diff
@@ -33,14 +32,15 @@ const LoggedInHandler = (() => {
             }
 
             // send marketing details to the backend, like when we open a new virtual account
-            // @see src/javascript/app/pages/user/new_account/virtual_acc_opening.js:82
+            // @see src/javascript/app/pages/user/new_account/virtual_acc_opening.js:bindValidation()
             const signup_device = LocalStore.get('signup_device') || /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent) ? 'mobile' : 'desktop';
-            const date_first_contact = LocalStore.get('date_first_contact') || toISOFormat(moment());
-            BinarySocket.send({
-                'set_settings'      : 1,
-                'signup_device'     : signup_device,
-                'date_first_contact': date_first_contact,
-            }).then(() => {
+            const date_first_contact = LocalStore.get('date_first_contact');
+            const payload = {
+                'set_settings' : 1,
+                'signup_device': signup_device,
```

Comment:
you can directly do it in the request object, e.g. `...(date_first_contact && { date_first_contact })`